OloEngine/src/OloEngine/Renderer/AnimatedModel.cpp

```
@@ -22,6 +247,56 @@ namespace OloEngine
+
+                Ref<Skeleton> skeleton;
+                if (SplitCombinedMeshSource(cachedMesh, m_Meshes, skeleton))
+                {
+                    m_Skeleton = skeleton;
+                    m_Animations = MeshCache::LoadAnimationsFromCache(sourcePath);
+
+                    // Load materials from the source file (lightweight Assimp read — no geometry postprocessing).
+                    // ProcessNode sets materialIndex = mesh->mMaterialIndex, so m_Materials[i]
+                    // must hold the material for the i-th Assimp mesh.
+                    {
+                        Assimp::Importer matImporter;
```

Clang (14.0.6)

[warning] variable 'matImporter' is not initialized

(cppcoreguidelines-init-variables)